datalad_next/itertools/decode_bytes.py

```diff
@@ -103,11 +103,8 @@ def handle_decoding_error(position: int,
         else:
             return (
                 position + exc.end,
-                joined_data[:position + exc.start].decode(encoding)
-                + joined_data[position + exc.start:position + exc.end].decode(
-                    encoding,
-                    errors='backslashreplace'
-                ),
+                joined_data[:position + exc.end].decode(
+                    encoding, errors='backslashreplace')
```

Comment:
I think this does not work. The code duplicates input data, as in the following example:

```python
>>> tuple(decode_bytes([b'08 War \xaf No \xaf More \xaf Trouble.shn.mp3'])
('08 War \\xaf', '08 War \\xaf No \\xaf', '08 War \\xaf No \\xaf More \\xaf', ' Trouble.shn.mp3')
```

I think the return statement should be (a missing `position` index is added):
```python
            return (
                position + exc.end,
                joined_data[position:position + exc.start].decode(encoding)
                + joined_data[position + exc.start:position + exc.end].decode(
                    encoding,
                    errors='backslashreplace'
                ),
            )
```